server/main.go

```diff
@@ -100,9 +101,10 @@ func Start() {
 
 	wl, _ = wal.NewNullWAL()
 	if config.Config.EnableWAL {
-		_wl, err := wal.NewAOFWAL(config.Config.WALDir)
+		WALDir := filepath.Join(config.DefaultParentDir, config.Config.WALDir)
```

Comment:
WALFile would not be in `parentDir`. It would always be in `/var/log/`. Check for similar paths in other OS arch. I think most people will use Mac and Unix, so it is safe to assume that.

Also, why the path on Mac should be `... Applications`, why can't it be `/var/xxxx`?

Reply:
Removed the WALFile dir
`/var/xxx` is generally protected for system data hence on macos it requires elevated privileges.